src/sentry/api/bases/organization_events.py

```diff
@@ -290,6 +293,15 @@ def build_cursor_link(self, request: Request, name: str, cursor: Optional[Cursor
             has_results="true" if bool(cursor) else "false",
         )
 
+    def handle_on_demand(self, request: Request) -> tuple[bool, MetricSpecType]:
+        use_on_demand_metrics = request.GET.get("useOnDemandMetrics") == "true"
+        on_demand_metric_type = MetricSpecType.SIMPLE_QUERY
+        on_demand_metric_type_value = request.GET.get("onDemandType")
+        if use_on_demand_metrics and on_demand_metric_type_value:
+            on_demand_metric_type = MetricSpecType(on_demand_metric_type_value)
```

Comment:
This will raise ValueError if the UI sends the wrong value which I can see being handled by the caller. Good to document the raising in the docstring.